### PR TITLE
Added testing for registration, added register form fields, fixed IntregityError for register

### PR DESCRIPTION
**Pull request by skydonline**
**Changes made:**
- Created `test_registration.py`, which tests the register endpoint.
  - Tests for a successful redirect (status code 302) and getting a successful response (status code 200) back. This was made for in the case of proper inputs in the register fields, as well as improper inputs (e.g. not matching passwords).
  - Checked flash messages to ensure the endpoint provides the appropriate flash messages
- Added additional fields to the `/register` endpoint, to ensure all the data is collected.
- Fixed #37 `IntegrityError` by modifying the user model.
  - Added a "Not Available" option to `GradeEnum`, which will be the default field in the case that the user registers without inputting a grade.

**Things to note:**
- Will modify `/register` logic in the future, to correctly input data if it is provided (like create the user model with a date of birth if it is given, otherwise leave it `Null`)
- Updated `result` for `passwordStrength.py` to have the list store the items as a string, to make it easier to work with across the codebase in the future

### CODE REVIEW

File: tests/test_registration.py [hunk 1]
```diff
@@ -1,7 +1,68 @@
...
+@pytest.fixture
+def client():
+    app.config['TESTING'] = True
+    with app.test_client() as client:
+        yield client
 
-# TODO
\ No newline at end of file
+def test_register(client):
+    response1 = client.post('/register', data={
```
Comment (MithunPara) [inline tests/test_registration.py:13]: I think some general comments explaining each client request could be helpful just for clarification

Comment (MithunPara) [inline tests/test_registration.py:23]: I think some general comments explaining each client request could be helpful just for clarification

File: tests/test_registration.py [hunk 2]
```diff
@@ -1,7 +1,68 @@
...
+        'name': 'James Smith',
+        'username': 'jsmith',
+        'date_of_birth': '2023-10-01',
+        'grade': '',
+        'email': 'jsmith99@gmail.com',
+        'password': 'jSmith123-',
+        'confirm_password': 'jSmith123-'
+    })
+    assert response2.status_code == 302
+    assert response2.location == '/register'
```
Comment (MithunPara) [inline tests/test_registration.py:77]: I believe that the location field for a response provides the full URL (could be mistaken). 
Using the lines below could parse the URL for the path:
from urllib.parse import urlparse
parsed_location = urlparse(response2.location)
assert parsed_location.path == '/register'

File: tests/test_registration.py [hunk 3]
```diff
@@ -1,7 +1,68 @@
...
+        'date_of_birth': '2023-10-01',
+        'grade': '',
+        'email': 'jsmith99@gmail.com',
+        'password': 'jSmith123-',
+        'confirm_password': 'differentPassword-'
+    })
+    assert response3.status_code == 302
+    assert response3.location == '/register'
+    
+    response4 = client.delete('/register', data={
```
Comment (MithunPara) [inline tests/test_registration.py]: Liked these assertions checking requests that should not be allowed for registration.

File: tests/test_registration.py [hunk 4]
```diff
@@ -1,7 +1,68 @@
 import pytest
+from app.src.app import app
 from app.src.auth import register
 from app.src.models import User, db
 
-from flask import Flask, request, redirect, url_for
+@pytest.fixture
```
Comment (MithunPara) [inline tests/test_registration.py:7]: Could add a temporary database in the fixture or just make sure to clear all the entries in the existing database after testing, to check whether a registered user is successfully stored in the database  (Lines that could be added to initialize database below):
with app.test_client() as client:
            with app.app_context():
                        db.create_all()
                        yield client
                       db.drop_all()

File: tests/test_registration.py [hunk 5]
```diff
@@ -1,7 +1,68 @@
...
+def client():
+    app.config['TESTING'] = True
+    with app.test_client() as client:
+        yield client
 
-# TODO
\ No newline at end of file
+def test_register(client):
```
Comment (MithunPara) [inline tests/test_registration.py]: Maybe add some extra assertions just for edge cases like an empty username.

**Review by MithunPara — CHANGES_REQUESTED** [review]

File: tests/test_registration.py (hunk 2 above)
Comment (skydonline) [reply tests/test_registration.py:77]: This checks the relative URL, not the full URL (i.e. http://127.0.0.1:5000/register = "/register")

**Review by racheldennis — APPROVED** [review]

**Review by MithunPara — APPROVED** [review]
